Source/WTF/wtf/MallocCommon.h

```diff
@@ -0,0 +1,246 @@
+    m_data = nullptr;
+    return data;
+}
+
+class ForbidMallocUseForCurrentThreadScope {
+public:
+#if ASSERT_ENABLED
+    WTF_EXPORT_PRIVATE ForbidMallocUseForCurrentThreadScope();
+    WTF_EXPORT_PRIVATE ~ForbidMallocUseForCurrentThreadScope();
+#else
+    ForbidMallocUseForCurrentThreadScope() = default;
+    ~ForbidMallocUseForCurrentThreadScope() { }
+#endif
+
+    ForbidMallocUseForCurrentThreadScope(const ForbidMallocUseForCurrentThreadScope&) = delete;
+    ForbidMallocUseForCurrentThreadScope(ForbidMallocUseForCurrentThreadScope&&) = delete;
+    ForbidMallocUseForCurrentThreadScope& operator=(const ForbidMallocUseForCurrentThreadScope&) = delete;
+    ForbidMallocUseForCurrentThreadScope& operator=(ForbidMallocUseForCurrentThreadScope&&) = delete;
+};
+
+class DisableMallocRestrictionsForCurrentThreadScope {
+public:
+#if ASSERT_ENABLED
+    WTF_EXPORT_PRIVATE DisableMallocRestrictionsForCurrentThreadScope();
+    WTF_EXPORT_PRIVATE ~DisableMallocRestrictionsForCurrentThreadScope();
+#else
+    DisableMallocRestrictionsForCurrentThreadScope() = default;
+    ~DisableMallocRestrictionsForCurrentThreadScope() { }
+#endif
+
+    DisableMallocRestrictionsForCurrentThreadScope(const DisableMallocRestrictionsForCurrentThreadScope&) = delete;
+    DisableMallocRestrictionsForCurrentThreadScope(DisableMallocRestrictionsForCurrentThreadScope&&) = delete;
+    DisableMallocRestrictionsForCurrentThreadScope& operator=(const DisableMallocRestrictionsForCurrentThreadScope&) = delete;
+    DisableMallocRestrictionsForCurrentThreadScope& operator=(DisableMallocRestrictionsForCurrentThreadScope&&) = delete;
+};
+
+#if ASSERT_ENABLED
+WTF_EXPORT_PRIVATE void assertMallocRestrictionForCurrentThreadScope();
+#else
+inline void assertMallocRestrictionForCurrentThreadScope() { }
+#endif
+
+} // namespace WTF
+
+using WTF::TryMallocReturnValue;
+using WTF::DisableMallocRestrictionsForCurrentThreadScope;
+using WTF::ForbidMallocUseForCurrentThreadScope;
+
+#define WTF_MAKE_CONFIGURABLE_ALLOCATED_IMPL(alloc, family) \
+    void* operator new(size_t, void* p) { return p; } \
+    void* operator new[](size_t, void* p) { return p; } \
+    \
+    void* operator new(size_t size) \
+    { \
+        return alloc::malloc(size); \
+    } \
+    \
+    void operator delete(void* p) \
+    { \
+        alloc::free(p); \
+    } \
+    \
+    void* operator new[](size_t size) \
+    { \
+        return alloc::malloc(size); \
+    } \
+    \
+    void operator delete[](void* p) \
+    { \
+        return alloc::free(p); \
+    } \
+    void* operator new(size_t, NotNullTag, void* location) \
+    { \
+        ASSERT(location); \
+        return location; \
+    } \
+    static void freeAfterDestruction(void* p) \
+    { \
+        alloc::fastFree(p); \
+    } \
+    using WTFIs ## family ## Allocated  = int; \
+
+#if ENABLE(MALLOC_HEAP_BREAKDOWN)
+#define WTF_MAKE_CONFIGURABLE_ALLOCATED_WITH_HEAP_IDENTIFIER_IMPL(classname, alloc, family) \
```

Comment:
Ditto, for `WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER_IMPL`.

Reply:
Discussed elsewhere, will do in another PR + add a FIXME here